go/vt/sqlparser/utils.go

```diff
@@ -156,7 +173,7 @@ func ReplaceTableQualifiers(query, olddb, newdb string) (string, error) {
 	// execute a query which slightly differs from the parsed
 	// version: e.g. 'where id=1' becomes 'where id = 1'.
 	if modified {
-		return String(upd), nil
+		return CanonicalString(upd), nil
```

Comment:
This is the cause of the vstream test failures. It causes the innodb_system tablespace name to be upper-cased which is not valid:
```
mysql> CREATE TABLE `vt_sharded`.`customer` (id int) TABLESPACE INNODB_SYSTEM;
ERROR 3510 (HY000): Tablespace INNODB_SYSTEM doesn't exist.
mysql> CREATE TABLE `vt_sharded`.`customer` (id int) TABLESPACE innodb_system;
Query OK, 0 rows affected (0.00 sec)
```

That innodb_system tablespace usage in the tests was added here: https://github.com/vitessio/vitess/pull/12672

Reply:
I added a new unit test case in `TestCanonicalOutput` that highlights the problem:
https://github.com/vitessio/vitess/pull/13866/commits/efebccb29c86869ac71aaf8097d58b02b5b613f1#diff-41f836d2a8dea7d5517b[…]da89a8c42c153b8232b7e4ef0

```
$ go test -timeout 30s -run ^TestCanonicalOutput$ vitess.io/vitess/go/vt/sqlparser

--- FAIL: TestCanonicalOutput (0.01s)
    --- FAIL: TestCanonicalOutput/create_table_a_(v_varchar(32))_engine=InnoDB_tablespace_innodb_system (0.00s)
        /Users/matt/git/vitess/go/vt/sqlparser/tracked_buffer_test.go:287:
                Error Trace:    /Users/matt/git/vitess/go/vt/sqlparser/tracked_buffer_test.go:287
                Error:          Not equal:
                                expected: "CREATE TABLE `a` (\n\t`v` varchar(32)\n) ENGINE InnoDB,\n  TABLESPACE innodb_system"
                                actual  : "CREATE TABLE `a` (\n\t`v` varchar(32)\n) ENGINE InnoDB,\n  TABLESPACE INNODB_SYSTEM"

                                Diff:
                                --- Expected
                                +++ Actual
                                @@ -3,2 +3,2 @@
                                 ) ENGINE InnoDB,
                                -  TABLESPACE innodb_system
                                +  TABLESPACE INNODB_SYSTEM
                Test:           TestCanonicalOutput/create_table_a_(v_varchar(32))_engine=InnoDB_tablespace_innodb_system
                Messages:       bad serialization
FAIL
FAIL    vitess.io/vitess/go/vt/sqlparser        0.523s
FAIL
```